File: day8/count_complete_treenode.java
class Solution {
    public int countNodes(TreeNode root) {
        if(root==null) return 0;
        // int cntl=0;
        // int cntr=0;
        // TreeNode tempr=root;
        // TreeNode temp=root;
        // while(temp.left!=null){
        //     cntl++;
        // }
        // while(tempr.right!=null){
        //     cntr++;
        // }
        // int ans=cntl+cntr;
        // return ans;
        return (1+countNodes(root.left)+countNodes(root.right));
    }
}

File: Count Complete Tree Node.py
#222. Count Complete Tree Nodes

from typing import Optional


class Solution:
    def countNodes(self, root: Optional[TreeNode]) -> int: # pyright: ignore[reportUndefinedVariable]
        
        if root is None:
            return 0
        
        left = self.countNodes(root.left)
        right = self.countNodes(root.right)
        
        return 1 + left + right